runtime-components/non-repudiation-perf/src/main/scala/com/daml/nonrepudiation/perf/NonRepudiationProxyBenchmark.scala

```diff
@@ -0,0 +1,167 @@
+
+import java.time.Clock
+import java.util.UUID
+
+import cats.effect.{ContextShift, IO}
+import com.daml.doobie.logging.Slf4jLogHandler
+import com.daml.ledger.api.v1.command_submission_service.CommandSubmissionServiceGrpc.CommandSubmissionServiceBlockingStub
+import com.daml.ledger.api.v1.command_submission_service.{
+  CommandSubmissionServiceGrpc,
+  SubmitRequest,
+}
+import com.daml.ledger.api.v1.commands.{Command, Commands, CreateCommand}
+import com.daml.ledger.api.v1.value.{Record, RecordField, Value}
+import com.daml.nonrepudiation.client.SigningInterceptor
+import com.daml.nonrepudiation.postgresql.Tables
+import com.daml.nonrepudiation.resources.HikariTransactorResourceOwner
+import com.daml.nonrepudiation.{AlgorithmString, NonRepudiationProxy}
+import com.daml.resources.Resource
+import com.daml.resources.grpc.{GrpcResourceOwnerFactories => Resources}
+import com.daml.testing.postgresql.{PostgresAround, PostgresDatabase}
+import com.google.protobuf.duration.Duration
+import doobie.util.log.LogHandler
+import io.grpc.Server
+import io.grpc.inprocess.{InProcessChannelBuilder, InProcessServerBuilder}
+import io.grpc.protobuf.services.ProtoReflectionService
+import org.openjdk.jmh.annotations._
+import sun.security.tools.keytool.CertAndKeyGen
+import sun.security.x509.X500Name
+
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.{Await, ExecutionContext}
+
+@State(Scope.Benchmark)
+class NonRepudiationProxyBenchmark extends PostgresAround {
+
+  import NonRepudiationProxyBenchmark._
+
+  @Param(Array("100000"))
+  var commandPayloadSize: Int = _
+
+  private var stub: CommandSubmissionServiceBlockingStub = _
+  private var proxy: Resource[ExecutionContext, Server] = _
+  private var database: PostgresDatabase = _
+  private var payload: String = _
+
+  @Benchmark
+  def run(): Unit = {
+    // Generating commands adds very little noise to substantial benchmarks
+    val command = generateCommand(payload)
+    val _ = stub.submit(command)
+  }
+
+  @Setup
+  def setup(): Unit = {
+    // The global fork-join work-stealing pool should be good enough to be used for both
+    // handling resource callbacks and mock command submission calls
+    implicit val executionContext: ExecutionContext = ExecutionContext.global
+    implicit val shift: ContextShift[IO] = IO.contextShift(executionContext)
+    implicit val logHandler: LogHandler = Slf4jLogHandler(classOf[NonRepudiationProxyBenchmark])
+    connectToPostgresqlServer()
+    database = createNewRandomDatabase()
+
+    val generator = new CertAndKeyGen(AlgorithmString.RSA, AlgorithmString.SHA256withRSA)
+    generator.generate(2048)
+    val key = generator.getPrivateKey
+    val certificate = generator.getSelfCertificate(
+      new X500Name("CN=Non-Repudiation Test,O=Digital Asset,L=Zurich,C=CH"),
+      1.hour.toSeconds,
+    )
+
+    val service = DummyCommandSubmissionService.bind(executionContext)
+
+    val participantName = InProcessServerBuilder.generateName()
```

Comment:
does the in process server builder involve a network connection (to localhost) or does it bypass that completely? If not, I think it might make sense to have a second benchmark which runs this closer to the scenario we recommend where those are run as separate processes (shouldn’t matter) communicating over a network connection (that should matter).

Reply:
> does the in process server builder involve a network connection (to localhost) or does it bypass that completely?

The latter.

I can add it for sure, but what would we be measuring in this way? The overhead of going through the network stack? Is it meaningful?

Reply:
I’d say it is meaningful. Less in the sense, that we can do much to reduce it so not super useful for driving optimizations but it is something that our clients will hit so in terms of the numbers they can expect, it probably comes closer when looking at the upper bound on what you could reach in a real setup.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/8916/commits/d107795813987a3fcd3fbfc6b7d3d72fa9b6c7cd